doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+Security Kernel Loader
+######################
+
+The Security Kernel Loader (SKL) is an implementation of a Secure Loader for
+the SKINIT CPU instruction. The SKL loader takes a minimal approach relying on
+the DRTM Preamble, e.g. the system bootloader, to load into memory and
+configure the Security Kernel that the SKL will measure and execute.
+
+Memory Layout
+*************
+
+The DRTM Preamble is responsible for allocating a contiguous block of physical
+memory and will ensure it is marked as reserved in the memory map. The
+necessary size for the allocation will be found in the SKL header. The DRTM
+Preamble will place the SKL at the beginning of the allocation.
+
+Below is the layout of how the memory allocation is segmented::
+
+      +---     Security Kernel Loader    ---+
+      |                                     |
+      v                                     v
+      +--------+----------+-------------+-------------+--------------+
+      | Header |   Code   |  Boot Tags  | IOMMU Table |  Log Buffer  |
+      +--------+----------+-------------+-------------+--------------+
+      ^                                 ^                            ^
+      |                                 |                            |
+      +                                 +                            +
+  sklbuf_base                          60kb                   header.skl_size
+
+
+SKL Header
+==========
+
+The SKL header is the AMD Secure Loader Block (SLB) header extended to meet the
+needs of the SKL. The first two fields are mandated by the SLB and the SKL adds
+two additional fields which can be seen in the following representative struct,
+::
+
+    typedef struct __packed sl_header {
```

Comment:
Technically speaking, `__packed` is something we should not have here.  It doesn't affect the layout of the struct, other than reducing the alignment to 1 which prohibits you from writing code like `u32 *entry_point_ptr = &sl_header.entry_point;`

Reply:
Ack, can drop here and maybe even to a PR to drop it from the current code.